libs/pbd/pbd/stack_allocator.h

```diff
@@ -108,12 +112,12 @@ class /*LIBPBD_API*/ StackAllocator
 
 	bool operator== (StackAllocator const& a) const
 	{
-		return &_buf == &a._buf;
+		return _buf == a._buf;
```

Comment:
This does not seem right. Rather than copying data, the target will point to existing data.
Memory corruption will ensure when the copy or original is modified

Reply:
If both `_buf`s are in the same address, they have the same identity, and hence their underlying data will be equal. Notice that `_buf` here is not a `boost::aligned_storage` but a simple array. Without indexing, it is already a pointer.

Reply:
Doing some research, I realised that 'alignas' works not only with C-arrays but with `std::array` as well. I will use that to avoid this kind of confusion for the lack of value semantics.